sympy/core/tests/test_expr.py

```diff
@@ -2214,3 +2215,9 @@ def test_21494():
 
 def test_Expr__eq__iterable_handling():
     assert x != range(3)
+
+
+def test_issue_23120():
+    assert (x*oo).is_Mul
```

Comment:
`-oo` is handled in the same loop and `zoo` already handles this case so no additional tests were added beyond the one for `oo`.